src/Core/tests/Benchmarks/Benchmarks/SetterBenchmarker.cs

```diff
@@ -0,0 +1,56 @@
+﻿using BenchmarkDotNet.Attributes;
+using Microsoft.Maui.Controls;
+
+namespace Microsoft.Maui.Benchmarks
+{
+	[MemoryDiagnoser]
+	public class SetterBenchmarker
+	{
+		static readonly VisualStateGroupList _vsg = new VisualStateGroupList()
+		{
+			new VisualStateGroup
+			{
+				States =
+				{
+					new VisualState { Name = VisualStateManager.CommonStates.Normal },
+					new VisualState
+					{
+						Name = VisualStateManager.CommonStates.Disabled,
+						Setters =
+						{
+							new Setter { Property = VisualElement.OpacityProperty, Value = 0.5 }
+						}
+					}
+				}
+			}
+		};
+		
+		// Avoids the warning:
+		// The minimum observed iteration time is 10.1000 us which is very small. It's recommended to increase it to at least 100.0000 ms using more operations.
+		const int Iterations = 100;
+
+		[Benchmark]
+		public void SettersFromDifferentSources()
+		{
+			var label = new Label
+			{
+				Style = new Style(typeof(Label))
+				{
+					Setters =
+					{
+						new Setter { Property = Label.TextProperty, Value = "Style" },
+						new Setter { Property = VisualStateManager.VisualStateGroupsProperty, Value = _vsg }
+					}
+				}
```

Comment:
Can we store the `Style` as `static readonly` and reuse it in this benchmark? It will take the time spent creating a `Style` out of the timing?

Reply:
That's a good idea.
I've updated the benchmark as requested, though it changed only the memory consumption by 1kb.
Regarding CPU nothing really changed.